lib/teiserver/microblog/libs/poll_response_lib.ex

```diff
@@ -0,0 +1,163 @@
+defmodule Teiserver.Microblog.PollResponseLib do
+  @moduledoc false
+  use TeiserverWeb, :library_newform
+  alias Teiserver.Microblog.{PollResponse, PollResponseQueries}
+  alias Phoenix.PubSub
+
+  @doc """
+  Returns the list of poll_responses.
+
+  ## Examples
+
+      iex> list_poll_responses()
+      [%PollResponse{}, ...]
+
+  """
+  @spec list_poll_responses(list) :: [PollResponse]
+  def list_poll_responses(args \\ []) do
+    args
+    |> PollResponseQueries.query_poll_responses()
+    |> Repo.all()
+  end
+
+  @spec get_poll_response(non_neg_integer(), non_neg_integer()) ::
+          PollResponse | nil
+  def get_poll_response(user_id, post_id) do
+    PollResponseQueries.query_poll_responses(
+      where: [
+        user_id: user_id,
+        post_id: post_id
+      ],
+      limit: 1
+    )
+    |> Repo.one()
+  end
+
+  @doc """
+  Creates a poll_response.
+
+  ## Examples
+
+      iex> create_poll_response(%{field: value})
+      {:ok, %PollResponse{}}
+
+      iex> create_poll_response(%{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def create_poll_response(attrs \\ %{}) do
+    %PollResponse{}
+    |> PollResponse.changeset(attrs)
+    |> Repo.insert()
+    |> broadcast_create_poll_response
+  end
+
+  defp broadcast_create_poll_response({:ok, %PollResponse{} = poll_response}) do
+    spawn(fn ->
+      # We sleep this because sometimes the message is seen fast enough the database doesn't
+      # show as having the new data (row lock maybe?)
```

Comment:
This is deeply suspicious. `Repo.insert` should immediately commit, and any subsequent read should see the latest value, that's the C of ACID.

Reply:
In my experiments that was not the case. Happy to remove it if you want though.

Reply:
What kind of experiments? How did you test that?
Did you have a local genserver subscribed to the topic and observed the race there?
I’m quite suspicious because that kind of artificial delay usually mask a true race condition, and I’d rather fix the underlying race than waiting and praying.

But also, alternatively, since there's no consumer for these topics, we could also remove all of this code and call it a day, it's not particularly hard to put it back either.

Reply:
I ran it locally, if I had a separate tab open to the index page of the blog it'd error saying it got a message there was a new post but that post didn't exist in the database. That said, at that stage I'd not put in the functionality of saying what type of update it was so in hindsight we can remove this part.